pkg/metrics/chaos-controller-manager.go

```diff
@@ -93,11 +99,23 @@ func NewChaosControllerManagerMetricsCollector(store cache.Cache, registerer pro
 			Name: "chaos_controller_manager_chaos_workflows",
 			Help: "Total number of chaos workflows",
 		}, []string{"namespace"}),
+		EmittedEvents: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name: "chaos_controller_manager_emitted_event_total",
+			Help: "Total number of the emitted event by chaos-controller-manager",
+		}, []string{"type", "reason", "namespace"}),
+	}
+
+	if registerer != nil {
+		registerer.MustRegister(c)
 	}
-	registerer.MustRegister(c)
 	return c
 }
 
+// NewTestChaosControllerManagerMetricsCollector provides metrics collector for testing
+func NewTestChaosControllerManagerMetricsCollector() *ChaosControllerManagerMetricsCollector {
+	return NewChaosControllerManagerMetricsCollector(nil, nil)
+}
+
```

Comment:
How about move this function to pkg `/controllers/utils/test/xxx` as it is only for testing?